daemons/attrd/attrd_ipc.c

```diff
@@ -252,6 +252,76 @@ handle_missing_host(xmlNode *xml)
     }
 }
 
+/* Convert a single IPC message with a regex into one with multiple children, one
+ * for each regex match.
+ */
+static int
+expand_regexes(xmlNode *xml, const char *attr, const char *value, const char *regex)
+{
+    if (attr == NULL && regex) {
+        GHashTableIter aIter;
+        regex_t r_patt;
+
+        crm_debug("Setting %s to %s", regex, value);
+        if (regcomp(&r_patt, regex, REG_EXTENDED|REG_NOSUB)) {
+            regfree(&r_patt);
```

Comment:
I looked into this before and IIRC we should only call regfree() if regcomp() returns success

Reply:
That makes sense.  Fixed.